src/main.cpp

```diff
@@ -1,17 +1,183 @@
+            "uav_gps_position", 10,
+            std::bind(&WaypointNode::callback_position, this, _1));
+
+    // Create a subscription for the "uav_mission_progress" topic
+    mission_progress_subscription =
+        this->create_subscription<interfaces::msg::MissionProgress>(
+            "uav_mission_progress", 10,
+            std::bind(&WaypointNode::callback_mission_progress, this, _1));
+
+    // Create a publisher for the "uav_waypoint_command" topic
+    uav_waypoint_command_publisher =
+        this->create_publisher<interfaces::msg::UAVWaypointCommand>(
+            "uav_waypoint_command", 10);
+
+    // Initialize Event Loop
+    event_loop_timer = this->create_wall_timer(
+        std::chrono::milliseconds(event_loop_time_delta_ms),
+        std::bind(&WaypointNode::event_loop, this));
+}
+
 /**
- * @brief A class that represents a minimal node.
+ * @brief Checks if the first loop was already triggered and returns the result.
+ *
+ * This function checks the state of the first loop and returns false if it was
+ * already triggered, otherwise it returns true. After returning the state, it
+ * sets the first loop state to false.
+ *
+ * @return False if the first loop was already triggered, true otherwise.
  */
-class WaypointNode : public common_lib::CommonNode {
-   public:
-    /**
-     * @brief Constructs a new WaypointNode object.
-     *
-     * @param id The unique name for the node.
-     */
-    WaypointNode() : CommonNode("waypoint_node") {}
-};
+bool WaypointNode::get_state_first_loop() {
+    // Check if first loop was already triggered
+    if (!state_first_loop) return false;
+
+    // Set first loop to false and return true
+    state_first_loop = false;
+    return true;
+}
+
+/**
+ * Checks if the current mission is finished.
+ *
+ * @return true if the mission is finished, false otherwise.
+ */
+bool WaypointNode::current_mission_finished() {
+    if (mission_progress >= 1.0) {
+        mission_progress = 0.0;
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * @brief Executes the event loop for the WaypointNode.
+ *
+ * This function is responsible for executing the event loop of the WaypointNode
+ * class. It checks the node state and performs the corresponding actions based
+ * on the state. The possible states are:
+ * - init: Initializes the mode.
+ * - pre_wait_time: Wait a predefined time.
+ * - reach_cruise_height: Fly drone to the crusing height.
+ * - fly_to_waypoint: Executes the mode to fly to the waypoint.
+ * - reach_target_height: Fly drone to the target height.
+ * - post_wait_time: Wait a predefined time.
+ *
+ * If the node state is unknown, an error message is logged and a job_finished
+ * error message is sent.
+ */
+void WaypointNode::event_loop() {
+    if (!this->get_active()) return;
+
+    switch (get_node_state()) {
+        case init:
+            mode_init();
+            break;
+        case pre_wait_time:
```

Comment:
Maybe add comment here that this state is exited in the WaypointNode::callback_wait_time() function, as this is not obvious if one has not read all source code.

Reply:
Changed in https://github.com/THI-Drone/waypoint_package/pull/3/commits/5a49068ac86cdab12bbed671d77177ca7fee6d9a